### PR TITLE
New component hlk_sw16 relay module

### PR DESCRIPTION
**Pull request by jameshilliard**
**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#17855

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/hlk_sw16.markdown [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+title: "HLK-SW16 Relay Module"
+description: "Instructions on how to integrate HLK-SW16 relay into Home Assistant."
+date: 2018-10-29 20:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: hlktech.jpg
+ha_category: DIY
+ha_release: 0.XX
```
Comment (MartinHjelmare) [inline source/_components/hlk_sw16.markdown]: 0.84

File: source/_components/hlk_sw16.markdown [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      c:
+        name: relay1-c
+      d:
+        name: relay1-d
+      e:
+        name: relay1-e
+      f:
+        name: relay1-f
+
+  relay2:
```
Comment (MartinHjelmare) [inline source/_components/hlk_sw16.markdown]: The first config example should be minimal and only include required config values for a normal set up.

File: source/_components/hlk_sw16.markdown [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      d:
+        name: relay2-d
+      e:
+        name: relay2-e
+      f:
+        name: relay2-f
+```
+
+{% configuration %}
+relayname:
```
Comment (MartinHjelmare) [inline source/_components/hlk_sw16.markdown]: This should be a `map` type key with a list of possible keys.

File: source/_components/switch.hlk_sw16.markdown [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+title: "HLK-SW16 Relay Module"
+description: "Instructions on how to integrate HLK-SW16 relay switches into Home Assistant."
+date: 2018-10-29 20:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: hlktech.jpg
+ha_category: Switch
+ha_release: 0.XX
```
Comment (MartinHjelmare) [inline source/_components/switch.hlk_sw16.markdown]: 0.84

File: source/_components/hlk_sw16.markdown (hunk 3 above)
Comment (MartinHjelmare) [reply source/_components/hlk_sw16.markdown]: Look at the preview and we can see that it doesn't look right currently.

https://deploy-preview-7250--home-assistant-docs.netlify.com/components/hlk_sw16/

File: source/_components/hlk_sw16.markdown [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      description: The control port for the relays.
+      required: false
+      type: integer
+      default: 8080
+    switches:
+      description: The array that contains the relays, each must be a number between 0 and 9 or letter between a and f which each corresponds to a labled relay switch on the HLK-SW16.
+      required: true
+      type: map
+      keys:
+        name:
```
Comment (MartinHjelmare) [inline source/_components/hlk_sw16.markdown]: I think we're missing the relay number map here.

File: source/_components/switch.hlk_sw16.markdown [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+title: "HLK-SW16 Relay Module"
+description: "Instructions on how to integrate HLK-SW16 relay switches into Home Assistant."
+date: 2018-10-29 20:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: hlktech.jpg
+ha_category: Switch
+ha_release: 0.94
```
Comment (MartinHjelmare) [inline source/_components/switch.hlk_sw16.markdown]: 0.84

**Review by MartinHjelmare — DISMISSED** [review]

**Comment by jameshilliard** [comment]
Any idea what's up with the test failure?

**Comment by jameshilliard** [comment]
Tests seem to be passing after I rebased.

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**